### PR TITLE
Feature/feedback functionality for anonymous use

### LINKED ISSUE #3132: Usability issue in feedback functionality for anonymous user
**Description**:

Right now, on browsing feedback tab in an API profile, an anonymous user (not logged into the platform) doesn't get to see any option to add Feedback for an API.
This is crucial if an API developer is requesting from her/his potential consumers for backlog requirements or feedback about the documentation on an API s/he is designing.
1st time visitors would leave without giving any feedback, unless they see a feedback button or hint for signing in/up in order to add feedback to an API.
So some action prompt is needed from the platform to encourage users add their feedback.

Another possible case could be that, user might be reluctant to provide their feedback and avoid the hassle if registration/signing in the platform is mandatory to add feedback to an API.

_**Feedback view in non-logged in session**_
image

_**Feedback view in logged in session**_
image

**User Story**:

"As an API owner, I need options to prompt 1st time visitors to add feedback to my API so that I can collect backlog requirements and improvement suggestions"

**Possible Solutions**
1. Show Add Feedback button also for people not signed in into the platform.
On clicking the button a dialog would appear prompting the user to sign in/up to give feedback.

2.  Providing a simple email address field and text box to provide feedback in case users don't wish to sign in or register to the platform.
Only API owner would be able to know the email address, so that s/he can further contact the person.
Providing an email address won't sign up the user in the platform.

### PR DESCRIPTION
**Pull request by krashna-deligence**
Closes #3132 

## Changes
- Add a feedback button for anonymous user
- On click event redirect to login and signup page

## Developer checklist
This checklist is to be completed by the PR developer:

- [ ] Alternative solutions were compared/discussed before writing code
    - [ ] trade-offs with this solution are considered acceptable
- [ ] Code in this PR adheres to the [project styleguide](CONTRIBUTING.md)
- [ ] This pull request does not decrease project test coverage
- [ ] If the code changes existing database collection(s), migration has been written
- [ ] If UI texts are added or changed, all texts are internationalized

## Reviewer checklist
Reviewed by: @Nazarah @marla-singer 

This list is to be completed by the pull request reviewer:
- [ ] Code works as described/expected
- [ ] Code seems to be error free
    - [ ] no browser console errors visible
    - [ ] no server console errors visible
    - [ ] passes CI build
- [ ] Code is written in a way that promotes maintainability
    - [ ] easy to understand
    - [ ] well organized
    - [ ] follows project coding standards and conventions
    - [ ] well documented

### CODE REVIEW

**Comment by Nazarah** [comment]
Updated the latest change in your branch and tried to run it in localhost. The tooltip doesn't appear on mouse hover. Screenshot is attached.
feedback-tooltip

File: apinf_packages/core/lib/i18n/en.i18n.json [hunk 1]
```diff
@@ -823,6 +824,10 @@
           "Error report": "Error report",
           "Feature request": "Feature request"
         }
+      },
+      "email": {
```
Comment (saransh-dev) [inline apinf_packages/core/lib/i18n/en.i18n.json]: Please remove additional field 'email'

File: apinf_packages/feedback/client/feedback.html [hunk 2]
```diff
@@ -47,6 +47,20 @@ <h4 id="feedback-title">
...
       </div>
+      {{ else }}
+      <button
+            type="button"
+            class="hidden-sm hidden-xs btn btn-info btn-xs feedback-form-button pull-right"
+            id="add-feedback-withoutLogin" title={{_ "feedback_feedbackForm_tooltip" }}>
+              <i class="fa fa-paper-plane" aria-hidden="true"></i>&nbsp;
+              {{_ "feedback_feedbackForm_button" }}
+      </button>
+      <button
```
Comment (saransh-dev) [inline apinf_packages/feedback/client/feedback.html]: Please add tool-tip here also.

File: apinf_packages/feedback/client/feedback.html [hunk 3]
```diff
@@ -47,6 +47,20 @@ <h4 id="feedback-title">
           </button>
         </div>
       </div>
+      {{ else }}
+      <button
+            type="button"
```
Comment (marla-singer) [inline apinf_packages/feedback/client/feedback.html]: @krashna-deligence Why did you add two buttons? As I see there needs only one button

File: apinf_packages/feedback/client/feedback.js [hunk 4]
```diff
@@ -59,4 +59,8 @@ Template.apiFeedback.events({
     // Change the visibility of all api's feedbacks
     Meteor.call('changeAllFeedbacksVisibility', slug, true);
   },
+  'click #add-feedback-withoutLogin': function () {
+    // Show feedbackForm modal without Login
+    FlowRouter.go('signIn');
```
Comment (marla-singer) [inline apinf_packages/feedback/client/feedback.js]: Incorrect comment. The action `FlowRouter.go()` doesn't show any modal. The action redirects to Sing in page

**Change text of comment**

File: apinf_packages/feedback/client/feedback.html [hunk 5]
```diff
@@ -47,6 +47,20 @@ <h4 id="feedback-title">
...
+            class="hidden-sm hidden-xs btn btn-info btn-xs feedback-form-button pull-right"
```
Comment (marla-singer) [inline apinf_packages/feedback/client/feedback.html]: What do classes "hidden-sm" and "hidden-xs" mean? 

**Remove them**

File: apinf_packages/feedback/client/feedback.html (hunk 3 above)
Comment (marla-singer) [reply apinf_packages/feedback/client/feedback.html]: Well, I tested it on a small screen and got the idea about two buttons. 
In this case please **leave some comment to describe proposal of using classes "hidden-*" and the second button**

File: apinf_packages/feedback/client/feedback.html [hunk 6]
```diff
@@ -47,6 +47,20 @@ <h4 id="feedback-title">
...
+            id="add-feedback-withoutLogin" title={{_ "feedback_feedbackForm_tooltip" }}>
+              <i class="fa fa-paper-plane" aria-hidden="true"></i>&nbsp;
+              {{_ "feedback_feedbackForm_button" }}
+      </button>
+      <button
+            type="button"
+            class="hidden-md hidden-lg btn btn-info btn-md feedback-form-button pull-right"
+            id="add-feedback-withoutLogin">
```
Comment (marla-singer) [inline apinf_packages/feedback/client/feedback.html]: Add tooltip for this button as well

**Comment by marla-singer** [comment]
> Updated the latest change in your branch and tried to run it in localhost. The tooltip doesn't appear on mouse hover. Screenshot is attached.

@Nazarah Can't agree with you cuz tooltip works and needs more time to shown
screen shot 2017-11-30 at 13 26 03

**Comment by marla-singer** [comment]
@krashna-deligence 
1. Before committing your changes please run linter on your local machine using the command
`npm run lint` or `yarn lint` 

2. After fixing all requested changes, squash all your commits into one using `git rebase`

- `git rebase develop -i`
-  Mark all commits as "s" except for the first one, and save
screenshot at nov 30 13-36-03
- write a commit message
- `git push -f`

**Comment by Nazarah** [comment]
@marla-singer I tested from my local environment with pulling the latest changes in the branch today again.
Waited for 1 minute, tooltip didn't appear. 
If it is working from your side, good to go.

**Comment by marla-singer** [comment]
@Nazarah Why do you push branch? 👀 
screenshot at nov 30 14-05-38

**Comment by Nazarah** [comment]
@marla-singer accidentally clicked sync in GitHub.
interestingly, i ran localhost again after clicking the sync to check if there is any problem caused there.
When I went to check the feedback, the tooltip appeared  in no time.
I don't know if it is the problem with the meteor in my PC.

File: apinf_packages/feedback/client/feedback.html [hunk 7]
```diff
@@ -47,6 +47,20 @@ <h4 id="feedback-title">
           </button>
         </div>
       </div>
+      {{ else }}
+      <button
+            type="button"
+            class="hidden-sm hidden-xs btn btn-info btn-xs feedback-form-button pull-right"
+            id="add-feedback-withoutLogin" title={{_ "feedback_feedbackForm_tooltip" }}>
```
Comment (marla-singer) [inline apinf_packages/feedback/client/feedback.html]: I checked tooltips in other places of our application. 

1. Add attribut `data-toggle="tooltip"`
2. Rename attribute `title` to `data-original-title`

**Comment by marla-singer** [comment]
@krashna-deligence I looked at existing code detailed and understood that (didn't work a long time with feedback part and had no relevant info about this part)

I don't like that you just douplicate buttons. Leave only one set of buttons and make flexibale logic

1. If a user is login then display the modal form on clicking "Send feedback".
2. If a user is not login then redirect to Sign in page on clicking "Send feedback" and display tooltip text what you added 

You can check if a user is login:
```js
const userId = Meteor.userId();

if (userId) {
  // User is login
} else {
  // User isn't login
}
```

The next part is to display title "Feedback and report messages" for all users because now Anonymous user doesn't see the title
joxi_screenshot_1512042835403

File: apinf_packages/core/lib/i18n/en.i18n.json [hunk 8]
```diff
@@ -25,8 +25,6 @@
   "addApi_header": "Add an API",
   "addApi_enterApiDetails": "Enter API details",
   "addApi_separateLineTitle_or": "or",
-  "addApi_sectionTitle_importOpenApiSpecification": "Import OpenAPI Specification",
```
Comment (marla-singer) [inline apinf_packages/core/lib/i18n/en.i18n.json]: You resolved the conflicts incorrectly. These changes come from the last `develop`
